pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -30,11 +31,18 @@ import (
 type framework struct {
 	registry         Registry
 	nodeInfoSnapshot *cache.NodeInfoSnapshot
+	waitingPods      []*WaitingPod
```

Comment:
This should probably be a map from a pod UID to the pod pointer. A map allows faster look-up. For example, a plugin might want to store UIDs of some pods that it monitors and approve them once a certain condition is met. The plugin can use to UIDs to find the pods faster in the map.